### DISCUSSION TITLE
Firefox Profile and Fcitx

### DISCUSSION THREAD

**Discussion by opqriu** [netblue30/firejail#4426] [Q&A] [answered]
https://github.com/netblue30/firejail/wiki/Frequently-Asked-Questions#how-can-i-enable-fcitx

==========

**~$ sudo gedit /etc/firejail/firefox.profile**

```
dbus-user filter
dbus-user.talk org.freedesktop.portal.Fcitx
ignore dbus-user none
```

**Ctrl C, Ctrl V, Save**

==========

Hello? 
I want to input cjk in firejail firefox. 
But this way firejail firefox won't run. 
What am I doing wrong?

**Comment by rusty-snake** [comment, marked as answer]
**Fix: `ignore nodbus` for firejail < 0.9.64.** For firejail >= 0.9.64 look at the [FAQ](https://github.com/netblue30/firejail/wiki/Frequently-Asked-Questions#how-can-i-enable-fcitx).

----

Original post:

> sudo gedit /etc/firejail/firefox.profile

OT: Use `gedit admin:///path/to/file`

> **~$ sudo gedit /etc/firejail/firefox.profile**
>
> dbus-user filter
> dbus-user.talk org.freedesktop.portal.Fcitx
> ignore dbus-user none

Two mistakes here:
1. It needs to be `firefox.local` and not `firefox.profile`.
   General: Never edit `/etc/firejail/*.profile` files because. Either copy them to `~/.config/firejail` or (the preferred way) use `*.local`s.
2. These three lines are for profiles with `dbus-user none`, but firefox.profile uses `dbus-user filter`. Therefore you only need the `dbus-user.talk` line.


> ### [How can I enable fcitx?](https://github.com/netblue30/firejail/wiki/Frequently-Asked-Questions#how-can-i-enable-fcitx)
>
> _Depending on the dbus-policy_ of the profile you need to add different command _to its local_. If the dbus-policy is set to filter, it is enough to add `dbus-user.talk org.freedesktop.portal.Fcitx`. If it is set to none, you need to add
>
> ```
> dbus-user filter
> dbus-user.talk org.freedesktop.portal.Fcitx
> ignore dbus-user none
> ```

Sum up: Try this:

```console
$ mkdir -p ~/.config/firejail
$ cat >> ~/.config/firejail/firefox.local <<EOF
dbus-user.talk org.freedesktop.portal.Fcitx
EOF
```
(or change `~/.config/firejail` to `/etc/firejail` and use a root shell)

**Comment by opqriu** [comment]
~/.config/firejail/firefox.local
OR /etc/firejail/firefox.local

Hi?
thanks.
And sorry!
I did what you told me but it doesn't work

Here, GIF.
https://imgur.com/a/OvatE4k

And,
If I delete the firefox.local file, firejail firefox will run.

**Reply by opqriu** [reply]
org.freedesktop.[portal] ::: There is no portal directory in dconf-editor's org - freedesktop.

**Reply by rusty-snake** [reply]
> ~/.config/firejail/firefox.local OR /etc/firejail/firefox.local

Does not matter. One is system-wide and requires root, the other not.

> org.freedesktop.[portal] ::: There is no portal directory in dconf-editor's org - freedesktop.

dbus != dconf
You can use tools like d-feet, qdbusviewer, bustle, dbus-monitor, dbus-send, gdbus, qdbus, busctl to inspect D-Bus.

> Error: line 1 in /etc/firejail/firefox.local is invalid

Guess your firejail is to old. "Fine-grained D-Bus sandboxing" was added in firejail 0.9.64. You can either try to install a newer version (e.g. backports if you use debian) or allow D-Bus entirely (with all it's consequences) using `ignore nodbus`.

**Reply by opqriu** [reply]
Thank You !!!!! :)
